tests/robustness/client/watch.go

```diff
@@ -56,17 +62,18 @@ func CollectClusterWatchEvents(ctx context.Context, t *testing.T, clus *e2e.Etcd
 		}
 	}()
 	wg.Wait()
-	return reports
+	return reports, everFailed, nil
 }
 
 type WatchConfig struct {
 	RequestProgress bool
 }
 
 // watchUntilRevision watches all changes until context is cancelled, it has observed revision provided via maxRevisionChan or maxRevisionChan was closed.
-func watchUntilRevision(ctx context.Context, t *testing.T, c *RecordingClient, maxRevisionChan <-chan int64, cfg WatchConfig) {
+func watchUntilRevision(ctx context.Context, lg *zap.Logger, c *RecordingClient, maxRevisionChan <-chan int64, cfg WatchConfig) bool {
```

Comment:
Code is good and correct. Just a nit: Returning bool for failure doesn't seem to be very Go like. 

Couple of possible improvements, implementing any of them should be ok:
* Avoid negative logic, it's starts getting confusing when you start negating or doing logic on it. Prefer returning `success` over `failed`.
* Return just error, you can both log and return error.
```
err = errors.New("client didn't collect all events")
lg.Error("Failed watch until revision", zap.Error(err))
```

Reply:
Thank you. I went with the first option as the existing robustness test still uses it to indicate whether the test case fails.